python/dllib/src/bigdl/dllib/utils/spark.py

```diff
@@ -313,54 +313,70 @@ def stop_spark_standalone():
-                     "spark.executorEnv.PYTHONHOME": python_env,
-                     "spark.executor.extraLibraryPath": ld_path,
-                     "spark.executorEnv.LD_PRELOAD": preload_so,
-                     "spark.kubernetes.container.image": container_image})
-        if "spark.driver.host" not in conf:
-            conf["spark.driver.host"] = get_node_ip()
-        if "spark.driver.port" not in conf:
-            conf["spark.driver.port"] = random.randint(10000, 65535)
-        if "BIGDL_CLASSPATH" in os.environ:
-            zoo_bigdl_jar_path = os.environ["BIGDL_CLASSPATH"]
-        else:
-            zoo_bigdl_jar_path = get_zoo_bigdl_classpath_on_driver()
-        if "spark.executor.extraClassPath" in conf:
-            conf["spark.executor.extraClassPath"] = "{}:{}".format(
-                zoo_bigdl_jar_path, conf["spark.executor.extraClassPath"])
-        else:
-            conf["spark.executor.extraClassPath"] = zoo_bigdl_jar_path
+            archive = "{}#{}".format(penv_archive, executor_python_env)
 
-        sc = self.create_sc(submit_args, conf)
+            submit_args = "--master " + master + " --deploy-mode client"
+            submit_args = submit_args + " --archives {}".format(archive)
+            submit_args = submit_args + gen_submit_args(
+                driver_cores, driver_memory, num_executors, executor_cores,
+                executor_memory, extra_python_lib, jars)
+
+            conf = enrich_conf_for_spark(conf, driver_cores, driver_memory, num_executors,
+                                         executor_cores, executor_memory,
+                                         extra_executor_memory_for_ray)
+            py_version = ".".join(platform.python_version().split(".")[0:2])
+            preload_so = executor_python_env + "/lib/libpython" + py_version + "m.so"
+            ld_path = executor_python_env + "/lib:" + executor_python_env + "/lib/python" + \
+                py_version + "/lib-dynload"
+            if "spark.executor.extraLibraryPath" in conf:
+                ld_path = "{}:{}".format(ld_path, conf["spark.executor.extraLibraryPath"])
+            conf.update({"spark.cores.max": num_executors * executor_cores,
+                         "spark.executorEnv.PYTHONHOME": executor_python_env,
+                         "spark.executor.extraLibraryPath": ld_path,
+                         "spark.executorEnv.LD_PRELOAD": preload_so,
+                         "spark.kubernetes.container.image": container_image})
+            if "spark.driver.host" not in conf:
+                conf["spark.driver.host"] = get_node_ip()
+            if "spark.driver.port" not in conf:
+                conf["spark.driver.port"] = random.randint(10000, 65535)
+            if "BIGDL_CLASSPATH" in os.environ:
+                zoo_bigdl_jar_path = os.environ["BIGDL_CLASSPATH"]
+            else:
+                zoo_bigdl_jar_path = get_zoo_bigdl_classpath_on_driver()
```

Comment:
This is not correct, the classpath on driver is not the same as that on executor. Need to modify according to yarn and test some training example.

Reply:
You are right. 
Updated. zoo_bigdl_path_on_executor is `python_env/lib/python3.7/site-packages/bigdl/share/dllib/lib/bigdl-dllib-spark_3.1.2-0.14.0-SNAPSHOT-jar-with-dependencies.jar` now.